src/components/Tasks.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useState } from 'react';
+
+import Task from './Task';
+import TaskCreateForm from './TaskCreateForm';
+import { Task as TaskModel } from '../models/task';
+
+export const Tasks: React.FC = () => {
+
+  // Our states, let's keep it simple
+  const [tasks, setTasks] = useState<TaskModel[]>([]);
+
+  const onCreated = (newTask: TaskModel): void => {
+    setTasks([newTask, ...tasks]);
+  }
+  const onDeleted = (id: number): void => {
+    setTasks(tasks.filter(t => t.id !== id));
+  }
+  const onUpdated = (id: number, updatedTask: TaskModel): void => {
+    setTasks([...tasks.map(t => ({ ...(t.id === id ? updatedTask : t) }))]);
```

Comment:
You don't need to spread `tasks` if you are mapping it. `map` already returns a new array.

Reply:
Correct, it would have made it slightly easier to read as well 👍 will clear it up when adding the tests for the edit functionality.